Source/Adobe.Target.Client/Model/TargetAttributes.cs

```diff
@@ -0,0 +1,235 @@
+        /// <param name="defaultValue">Default attribute value</param>
+        /// <returns>String attribute or default value</returns>
+        public string GetString(string mbox, string key, string defaultValue = default)
+        {
+            var mboxDict = this.ToMboxDictionary(mbox);
+            if (mboxDict == null)
+            {
+                return defaultValue;
+            }
+
+            return mboxDict.TryGetValue(key, out var value) && value != null ? value.ToString() : defaultValue;
+        }
+
+        /// <summary>
+        /// Gets an Integer Attribute
+        /// </summary>
+        /// <param name="mbox">Mbox name</param>
+        /// <param name="key">Attribute name</param>
+        /// <param name="defaultValue">Default attribute value</param>
+        /// <returns>Integer attribute or default value</returns>
+        public int GetInteger(string mbox, string key, int defaultValue = default)
+        {
+            return this.GetValue(mbox, key, defaultValue);
+        }
+
+        /// <summary>
+        /// Gets a Double Attribute
+        /// </summary>
+        /// <param name="mbox">Mbox name</param>
+        /// <param name="key">Attribute name</param>
+        /// <param name="defaultValue">Default attribute value</param>
+        /// <returns>Double attribute or default value</returns>
+        public double GetDouble(string mbox, string key, double defaultValue = default)
+        {
+            return this.GetValue(mbox, key, defaultValue);
+        }
+
+        /// <summary>
+        /// Gets an Attribute value
+        /// </summary>
+        /// <typeparam name="T">Generic type parameter</typeparam>
+        /// <param name="mbox">Mbox name</param>
+        /// <param name="key">Attribute name</param>
+        /// <param name="defaultValue">Default attribute value</param>
+        /// <returns>Attribute or default value</returns>
+        public T GetValue<T>(string mbox, string key, T defaultValue = default)
+        {
+            var mboxDict = this.ToMboxDictionary(mbox);
+            if (mboxDict == null)
+            {
+                return defaultValue;
+            }
+
+            try
+            {
+                return mboxDict.TryGetValue(key, out var value) && value != null
+                    ? (T)Convert.ChangeType(value, typeof(T)) : defaultValue;
+            }
+            catch (Exception)
+            {
+                return defaultValue;
+            }
+        }
+
+        private static IReadOnlyDictionary<string, IReadOnlyDictionary<string, object>> ToDictionary(TargetDeliveryResponse deliveryResponse)
+        {
+            if (deliveryResponse?.Response == null)
+            {
+                return null;
+            }
+
+            var result = new Dictionary<string, Dictionary<string, object>>();
+            var prefetchResponse = deliveryResponse.Response.Prefetch;
+            var executeResponse = deliveryResponse.Response.Execute;
```

Comment:
I based it on the way it's implemented in JavaSDK, let me trey to refactor as suggested..

Reply:
I can't remove passing down the common accumulator as it will introduce inconsistencies with JavaSDK in what regards order of option adding, and thus which content is actually present in values in case of equal keys. For now, I'll just make the methods return the accumulator.

Reply:
@XDex I am not sure I follow your argument, the dictionary keys and values are unordered, but I am OK with your latest changes.